code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -57,10 +57,12 @@
 	if(response == "Sync")
 		var/success = FALSE
 		for(var/obj/machinery/r_n_d/server/S in SSmachinery.all_machines)
-			for(var/datum/tech/T in files.known_tech) //Uploading
+			for(var/id in files.known_tech) //Uploading
+				var/datum/tech/T = files.known_tech[id]
 				S.files.AddTech2Known(T)
-			for(var/datum/tech/T in S.files.known_tech) //Downloading
-				files.AddTech2Known(T)
+			for(var/id in S.files.known_tech) //Downloading
```

Comment:
If we already uploaded, doesn't `S.files.known_tech` now contain the complete list of known_tech at highest researched levels across the machine and the server? So we can just copy the list instead of doing this second loop?

So we can just have this:
```
			files.known_tech = S.files.known_tech.copy()
```